scanners/zap/zap.py

```diff
@@ -649,35 +650,16 @@ def _setup_passive_wait(self):
     def _setup_replacer(self):
         """Adds the replacer to the job list"""
 
-        def _validate_rule_boolean_values(rule):
-            if not isinstance(rule["matchRegex"], bool):
-                raise ValueError("The matchRegex in the replacer rule must be set to a Boolean value")
-
-            if "tokenProcessing" in rule and not isinstance(rule["tokenProcessing"], bool):
-                raise ValueError("The tokenProcessing in the replacer rule must be set to a Boolean value")
-
-        if not self.my_conf("replacer"):
+        if not self.cfg.replacer:
             return
 
-        rules = self.my_conf("replacer.rules")
-        if rules:
```

Comment:
This validation is already handled by dataclasses/dacite, so it has been removed from here